Core/Src/main.c

```diff
@@ -87,17 +89,41 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
-
+  HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);  // Start PWM on channel 1
+  uint8_t rxData;
+  uint8_t txData;
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+	// Set the chip select low to start communication
+	HAL_GPIO_WritePin(GPIOA, GPIO_PIN_8, GPIO_PIN_RESET);
+
+	txData = 1;
+	// First transmission to send start bit
+	HAL_SPI_TransmitReceive(&hspi1, txData, rxData, 1, HAL_MAX_DELAY);
+	txData = 128;
+	// second transmission to send control bits and receive first few adc values
+	HAL_SPI_TransmitReceive(&hspi1, txData, rxData, 1, HAL_MAX_DELAY);
+	uint16_t adc_value_1 = rxData << 8;
+	// third transmission to recieve 8 adc bits
+	HAL_SPI_TransmitReceive(&hspi1, txData, rxData, 1, HAL_MAX_DELAY);
+	uint16_t adc_value = adc_value_1 | rxData;
+
+	// Set the chip select high to end communication
+	HAL_GPIO_WritePin(GPIOA, GPIO_PIN_8, GPIO_PIN_SET);
+
+	uint16_t pwm_value = (adc_value*7.5) / 1024;
```

Comment:
pwm math isn't correct. Make sure you divide the adc value by its max counts first, and you forgot one more thing... Hint: It has something to do with duty cycle.

Reply:
Also, The maximum numerical value for a (unsigned) 10-bit number is 1023. 1024? Nuh-uh!

Reply:
oh whoops, mb also wasnt sure about the duty cycle i though all i had to do was multiply by the duty cycle percentage and i chose 7.5 since thats between 5-10% but i pushed a new commit, i found a register that directly sets the duty cycle i think

Reply:
https://github.com/UWARG/embedded-bootcamp/pull/93/commits/db1e1b7760b002434d014b5e5c4048611aa86828 okay so I changed the prescaler and counter period value first i completely forgot to do that, and then i redid the pwm calculation to map the adc value to the pwm duty cycle bounds, is that how it should be done?